llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9131,10 +9129,22 @@ void LoopVectorizationPlanner::buildVPlansWithVPRecipes(ElementCount MinVF,
                                                         ElementCount MaxVF) {
   assert(OrigLoop->isInnermost() && "Inner loop expected.");
 
+  // Only use noalias metadata when using memory checks guaranteeing no overlap
+  // across all iterations.
+  const LoopAccessInfo *LAI = Legal->getLAI();
+  std::unique_ptr<LoopVersioning> LVer = nullptr;
```

Comment:
Can this be moved to Legal to avoid the unique_ptr? Probably not worth it if it's too much work.